### PR TITLE
Add "View source" and "File issue" buttons to each page

### LINKED ISSUE #696: Consider adding edit/issue links to each page
See https://github.com/flutter/flutter/issues/10416 & https://github.com/flutter/website/pull/592.

### PR DESCRIPTION
**Pull request by chalin**
Fixes #696, adding Edit and "File Issue" to all pages, except for the homepage and the error page.

Staged at https://angulardart-org-dev.firebaseapp.com, e.g.:
- https://angulardart-org-dev.firebaseapp.com/guides/get-started
- https://angulardart-org-dev.firebaseapp.com/angular/guide
- https://angulardart-org-dev.firebaseapp.com/components

This adds button to all pages 
This is what it will look like on narrow devices:

---

screen shot 2017-11-15 at 11 00 24

---

screen shot 2017-11-15 at 11 00 33

### CODE REVIEW

File: src/_includes/page-edit-btn-group.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% if page.long-title or page.title -%}
+<div class="page-edit-btn-group btn-group" role="group">
+  <a
+    href="{{site.repo}}/tree/{{site.branch}}/{{site.source}}/{{page.path}}"
+    class="btn btn-sm no-automatic-external"
+    title="Edit Source"
```
Comment (kwalrath) [inline src/_includes/page-edit-btn-group.html]: Source -> source

File: src/_includes/page-edit-btn-group.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    target="_blank" rel="noopener">
+    <i class="fa fa-pencil"></i>
+  </a>
+  <a
+    href="{{site.repo}}/issues/new?title='{{page.title}}' page issue&body=From URL: {{site.url}}{{page.url}}"
+    class="btn btn-sm no-automatic-external"
+    title="File Issue"
```
Comment (kwalrath) [inline src/_includes/page-edit-btn-group.html]: Issue -> issue

File: src/_includes/page-edit-btn-group.html (hunk 1 above)
Comment (kwalrath) [reply src/_includes/page-edit-btn-group.html]: But we agreed this should be "View source", since we won't take them directly to the editor.

Comment (chalin) [reply src/_includes/page-edit-btn-group.html]: done

File: src/_includes/page-edit-btn-group.html (hunk 2 above)
Comment (chalin) [reply src/_includes/page-edit-btn-group.html]: done

**Comment by chalin** [comment]
- Required updates are done.
- Also **fixed the indentation of Angular page titles** when **viewed on narrow-display** devices.

The staging server has been refreshed.

---

### Examples follow

screen shot 2017-11-15 at 18 12 27

screen shot 2017-11-15 at 18 12 21

screen shot 2017-11-15 at 18 11 19

screen shot 2017-11-15 at 18 10 41

---

### _Before_ the indentation fix ...

Before the indentation fix included in this PR, an Angular page title would have looked like this:

screen shot 2017-11-15 at 17 47 50

File: src/_includes/page-github-buttons.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    href="{{site.repo}}/tree/{{site.branch}}/{{site.source}}/{{page.path}}"
+    class="btn no-automatic-external"
+    title="View page source"
+    target="_blank" rel="noopener">
+    <i class="fa fa-github"></i>
+  </a>
+  <a
+    href="{{site.repo}}/issues/new?title='{{page.title}}' page issue&body=From URL: {{site.url}}{{page.url}}"
+    class="btn no-automatic-external"
+    title="File an issue over this page"
```
Comment (kwalrath) [inline src/_includes/page-github-buttons.html]: How about:

Report a bug on this page

File: src/_includes/page-github-buttons.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{% if page.long-title or page.title -%}
+<div class="page-edit-btn-group btn-group" role="group">
+  <a
+    href="{{site.repo}}/tree/{{site.branch}}/{{site.source}}/{{page.path}}"
+    class="btn no-automatic-external"
```
Comment (kwalrath) [inline src/_includes/page-github-buttons.html]: When I mouse down on one of the icons, a bit of gray appears at the left and top. I didn't expect that. Can you get rid of it?

Also, after you click an icon, it turns gray/black and doesn't turn blue again until (1) the window loses focus (but it'll turn black again when the window regains focus) or (1) you click the other icon. It took me a while to figure out what the pattern was!

These are minor issues and not worth holding up this feature for too long... but I'd like them fixed soon.

Comment (chalin) [reply src/_includes/page-github-buttons.html]: Oh! I hadn't noticed that. Strange behavior. I'll investigate and try to fix this.

**Comment by chalin** [comment]
I had originally imitated what is done on flutter.io, and used Bootstrap `btn` and `btn-group` styles, but the icons/links aren't really buttons in the way that bootstrap intends it. So I shift to a "horizonal menu" implementation, similar to what was already done in the footer.

The staging server has been updated.

**Comment by kwalrath** [comment]
Much better! (Although that weird color change is still happening.)

Strangely, now the icons are way bigger on non-angular-themed pages. E.g., here's Get Started:

image

They're still a bit small in the Angular pages:

image

Can we get something in between for both page styles?

**Pull request merged**